configs/configs.go

```diff
@@ -2,25 +2,55 @@ package configs
 
 import (
 	"os"
+	"path/filepath"
 
 	"github.com/LetsFocus/goLF/logger"
 	"github.com/joho/godotenv"
 )
 
-func NewConfig(log *logger.CustomLogger) Config {
-	env := os.Getenv("APP_ENV")
-	envPath := ""
-	if env != "" {
-		envPath = "./configs/." + env + ".env"
-	} else {
-		envPath = "./configs/.env"
+func findConfigsDir(dir string) (string, error) {
+	depth := 0
+	maxDepth := 3
+	configsDir := filepath.Join(dir, "configs")
+	if _, err := os.Stat(configsDir); err == nil {
+		return configsDir, nil
 	}
-	
-	if err := godotenv.Load(envPath); err != nil {
-		log.Error("No .env file found")
+	if depth < maxDepth {
+		parentDir := filepath.Dir(dir)
+		if parentDir == dir {
+			return "", os.ErrNotExist
+		}
+		depth++
+		findConfigsDir(parentDir)
+	}
+	return "", os.ErrNotExist
+}
+
+func NewConfig(log *logger.CustomLogger) Config {
+	currentDir, err := os.Getwd()
+	if err != nil {
+		log.Error("Unable to get current directory")
+		return Config{Log: log}
 	}
+	path, err := findConfigsDir(currentDir)
+	if err != nil {
```

Comment:
remove this if condition, make the else condition as if. and add this log before return. We don't need a if else ladder here, single if is sufficient.